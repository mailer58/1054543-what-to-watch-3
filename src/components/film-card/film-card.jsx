import React from 'react';
import PropTypes from "prop-types";

export const FilmCard = ({title, img, onMouseOverCard, onMouseOutCard}) => {
  return (<article onMouseOver={onMouseOverCard} onMouseOut={onMouseOutCard} className="small-movie-card catalog__movies-card">
    <div className="small-movie-card__image">
      <img src={img} alt={title} width="280" height="175" />
    </div>
    <h3 className="small-movie-card__title">
      <a className="small-movie-card__link" href="movie-page.html">{title}</a>
    </h3>
  </article>
  );
};

FilmCard.propTypes = {
  title: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  onMouseOverCard: PropTypes.func.isRequired,
  onMouseOutCard: PropTypes.func.isRequired
};
