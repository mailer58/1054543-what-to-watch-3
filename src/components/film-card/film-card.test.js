import React from "react";
import renderer from "react-test-renderer";
import {FilmCard} from "./film-card.jsx";

const films = {
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  cardImg: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
};

const film = {
  id: 1,
  title: `Aviator`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  scoring: 5,
  description: `text`,
  ratings: 100,
  director: `director`,
  starring: `actor`,
  genre: `Drama`,
  year: 2000,
  cardImg: `img/aviator.jpg`
};

it(`Should FilmCard render correctly`, () => {
  const tree = renderer
    .create(<FilmCard
      title = {films.title}
      img = {films.cardImg}
      onMouseOverCard = {jest.fn(film)}
      onMouseOutCard = {jest.fn()}
      onClickCard = {jest.fn()}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
