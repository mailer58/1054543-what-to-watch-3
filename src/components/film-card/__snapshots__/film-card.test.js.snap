// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmCard render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[MockFunction]}
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      src="http"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
